src/Cmarfil/LaravelMultiUserJsonSettings/Setting.php

```diff
@@ -0,0 +1,255 @@
+     */
+    protected $dirty = false;
+
+    /**
+     * Whether settings have been loaded from the database (this session).
+     *
+     * @var array
+     */
+    protected $loaded = array();
+
+
+    /**
+     * Construction method to read package configuration.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->table = \Config::get('cmarfil/laravel-multiuser-json-settings::table');
+        $this->column = \Config::get('cmarfil/laravel-multiuser-json-settings::column');
+        $this->custom_constraint = \Config::get('cmarfil/laravel-multiuser-json-settings::custom_constraint');
+		$this->constraint_key = \Config::get('cmarfil/laravel-multiuser-json-settings::constraint_key');
+		$this->default_constraint_value = \Config::get('cmarfil/laravel-multiuser-json-settings::constraint_key');
+    }
+
+
+    /**
+     * Get the value of a specific setting.
+     *
+     * @param string $key
+     * @param mixed $default
+     * @param mixed $custom_constraint_value
+     * @return mixed
+     */
+    public function get($key, $default = null, $custom_constraint_value = false)
+    {
+		$constraint_value = $this->negotiate_constraint_value($custom_constraint_value);
+        $this->check($constraint_value);
+
+        return array_get($this->settings[$constraint_value], $key, $default);
+    }
+
+    /**
+     * Set the value of a specific setting.
+     *
+     * @param string $key
+     * @param mixed $value
+	 * @param mixed $custom_constraint_value
+     */
+    public function set($key, $value = null, $custom_constraint_value = false)
+    {
+		$constraint_value = $this->negotiate_constraint_value($custom_constraint_value);
+        $this->check($constraint_value);
+
+        $this->dirty = true;
+
+        if (is_array($key)) {
+            foreach ($key as $k => $v) {
+                array_set($this->settings[$constraint_value], $k, $v);
+            }
+        } else {
+            array_set($this->settings[$constraint_value], $key, $value);
+        }
+    }
+
+    /**
+     * Unset a specific setting.
+     *
+     * @param string $key
+	 * @param mixed $custom_constraint_value
+     * @return void
+     */
+    public function forget($key, $custom_constraint_value = false)
+    {
+		$constraint_value = $this->negotiate_constraint_value($custom_constraint_value);
+        $this->check($constraint_value);
+
+        if (array_key_exists($key, $this->settings[$constraint_value])) {
+            unset($this->settings[$constraint_value]);
+
+            $this->dirty = true;
+        }
+    }
+
+    /**
+     * Check for the existance of a specific setting.
+     *
+     * @param string $key
+     * @return bool
+     */
+    public function has($key, $custom_constraint_value = false)
+    {
+		$constraint_value = $this->negotiate_constraint_value($custom_constraint_value);
+        $this->check($constraint_value);
+
+        return array_key_exists($key, $this->settings[$constraint_value]);
```

Comment:
This will throw "*array_key_exists() expects parameter 2 to be array, null given*" if the `$custom_constraint_value` references a row with no settings.

We might need to check for the existence of `$this->settings[$constraint_value]` first? So maybe something like:

```php
return array_key_exists($constraint_value, $this->settings) ?: array_key_exists($key, $this->settings[$constraint_value]);
```